internal/api/grpc/admin/integration_test/restrictions_allow_public_org_registrations_test.go

```diff
@@ -70,28 +71,34 @@ func awaitPubOrgRegDisallowed(t *testing.T, ctx context.Context, cc *integration
 // awaitGetSSRGetResponse cuts the CSRF token from the response body if it exists
 func awaitGetSSRGetResponse(t *testing.T, ctx context.Context, client *http.Client, parsedURL *url.URL, expectCode int) string {
 	var csrfToken []byte
-	await(t, ctx, func(tt *assert.CollectT) {
-		resp, err := client.Get(parsedURL.String())
-		require.NoError(tt, err)
-		body, err := io.ReadAll(resp.Body)
-		require.NoError(tt, err)
-		searchField := `<input type="hidden" name="gorilla.csrf.Token" value="`
-		_, after, hasCsrfToken := bytes.Cut(body, []byte(searchField))
-		if hasCsrfToken {
-			csrfToken, _, _ = bytes.Cut(after, []byte(`">`))
-		}
-		assert.Equal(tt, resp.StatusCode, expectCode)
-	})
+	retryDuration, tick := integration.WaitForAndTickWithMaxDuration(ctx, time.Minute)
+	require.EventuallyWithT(t,
+		func(tt *assert.CollectT) {
+			resp, err := client.Get(parsedURL.String())
+			require.NoError(tt, err)
+			body, err := io.ReadAll(resp.Body)
+			require.NoError(tt, err)
+			searchField := `<input type="hidden" name="gorilla.csrf.Token" value="`
+			_, after, hasCsrfToken := bytes.Cut(body, []byte(searchField))
+			if hasCsrfToken {
+				csrfToken, _, _ = bytes.Cut(after, []byte(`">`))
+			}
+			assert.Equal(tt, resp.StatusCode, expectCode)
+		}, retryDuration, tick, "awaiting successful callback failed",
```

Comment:
error messages are generic
(please see also the other changes)

Reply:
added error messages to every eventuallyWithT changed in this PR